src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -39,47 +41,47 @@ case class IllegalLabelKeyException(labelKey: String)
   extends LeoException(s"Labels cannot have a key of '$labelKey'", StatusCodes.NotAcceptable)
 
 
-class LeonardoService(protected val dataprocConfig: DataprocConfig, protected val clusterResourcesConfig: ClusterResourcesConfig, protected val proxyConfig: ProxyConfig, gdDAO: DataprocDAO, dbRef: DbReference, val clusterMonitorSupervisor: ActorRef)(implicit val executionContext: ExecutionContext) extends LazyLogging {
+class LeonardoService(protected val dataprocConfig: DataprocConfig,
+                      protected val clusterResourcesConfig: ClusterResourcesConfig,
+                      protected val proxyConfig: ProxyConfig,
+                      protected val petServiceAccountConfig: PetServiceAccountConfig,
+                      protected val gdDAO: DataprocDAO,
+                      protected val googleIamDAO: GoogleIamDAO,
+                      protected val dbRef: DbReference,
+                      protected val clusterMonitorSupervisor: ActorRef,
+                      protected val samDAO: SamDAO)
```

Comment:
This isn't really a review suggestion - more a comment: this is getting a bit crowded - unsure if we'll need to add many more parameters in the future though

Reply:
We could have a `LeonardoConfig` case class which contains sub-config case classes. Then we just pass around a `LeonardoConfig` instance instead of individual ones.. .just an idea.